mod/beacon/blockchain/types.go

```diff
@@ -139,6 +167,44 @@ type EventFeed[EventT any] interface {
 	Send(event EventT) int
 }
 
+// ExecutionPayload is the interface for the execution payload.
+type ExecutionPayload[T any, WithdrawalT Withdrawal] interface {
+	Empty(uint32) T
```

Comment:
i could be wrong, but i believe this level of constraint on ExecutionPayloadT isn't needed in this service